main.c

```diff
@@ -130,60 +228,30 @@ int main(int argc, char **argv)
     // index array via ascii of char, value is its count
     int char_count[CHARSET_SIZE] = {0};
 
-    FILE *f_ptr = fopen(f_name, "rb");
-    if (f_ptr == NULL)
+
+    FILE *fp = fopen(f_name, "rb");
+    if (fp == NULL)
     {
         fprintf(stderr, "Can't open %s. Exiting...\n", f_name);
         exit(1);
     }
 
-    // get file size:
-    fseek(f_ptr, 0, SEEK_END);
-    long f_size = ftell(f_ptr);
-    rewind(f_ptr);
-
-    long bytes_read = 0;
-
-    while (bytes_read < f_size)
-    {
-        int bytes_to_read = MAX_BUFFER_SIZE;
-
-        if (f_size - bytes_read < MAX_BUFFER_SIZE)
-            bytes_to_read = f_size - bytes_read;
-
-        // allocate memory to contain the whole file:
-        char *buffer = (char *)malloc(sizeof(char) * bytes_to_read);
-        if (buffer == NULL)
-        {
-            fprintf(stderr, "Couldn't allocate buffer. Exiting...\n");
-            fclose(f_ptr);
-            exit(1);
-        }
-
-        // copy the file into the buffer:
-        size_t result = fread(buffer, 1, bytes_to_read, f_ptr);
-        if (result != bytes_to_read)
-        {
-            fprintf(stderr, "Couldn't read from file. Exiting...\n");
-            free(buffer);
-            fclose(f_ptr);
-            exit(1);
-        }
-
-        bytes_read += bytes_to_read;
+    int c = fgetc(fp);
+    if (feof(fp)) {
+        // this text to stdout is needed for char count test1.txt
+        fprintf(stdout, "File is empty. Exiting...\n");
```

Comment:
Maybe print this message to stderr? I feel that an empty file should encode into an empty file

Reply:
^only applies if we decide to output the encoding to stdout (see comment below)